@here/olp-sdk-dataservice-read/lib/client/VersionedLayerClient.ts

```diff
@@ -250,11 +268,32 @@ export class VersionedLayerClient {
         request: QuadKeyPartitionsRequest | PartitionsRequest,
         abortSignal?: AbortSignal
     ): Promise<QueryApi.Index | MetadataApi.Partitions | QueryApi.Partitions> {
-        if (this.version === undefined) {
-            this.version = await this.getLatestVersion(
+        let usingVersion: number | undefined;
+        const requestVersion = request.getVersion();
+
+        if (requestVersion !== undefined) {
+            usingVersion = requestVersion;
+        } else {
+            usingVersion = this.version;
+        }
+
+        if (usingVersion === undefined) {
+            // fetch the latest version and lock it to the instance.
+            usingVersion = await this.getLatestVersion(
                 request.getBillingTag()
             ).catch(error => Promise.reject(error));
+
+            this.version = usingVersion;
         }
+
+        if (usingVersion === undefined) {
+            return Promise.reject(
+                new Error(
+                    `No version provided. Please provide version to the Request or lock version in the constructor`
+                )
+            );
+        }
```

Comment:
How can this be? This can only happen if the `getLatestVersion()` fails, yes? But then it is a different kind of error and not a `No version provided` error. Rather `Unable to retrieve latest version....`

Reply:
yes, you are right. changing

Reply:
updated